Twitch-Chat-Downloader: init with dependencies

strager opened the pull request:
<!-- Nixpkgs has a lot of new incoming Pull Requests, but not enough people to review this constant stream. Even if you aren't a committer, we would appreciate reviews of other PRs, especially simple ones like package updates. Just testing the relevant package/service and leaving a comment saying what you tested, how you tested it and whether it worked would be great. List of open PRs: <https://github.com/NixOS/nixpkgs/pulls>, for more about reviewing contributions: <https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#sec-reviewing-contributions>. Reviewing isn't mandatory, but it would help out a lot and reduce the average time-to-merge for all of us. Thanks a lot if you do! -->
###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @berdario @thanegill

strager committed:
pythonPackages.twitch-python: init at 0.0.17

I want to add Twitch-Chat-Downloader to Nixpkgs. Twitch-Chat-Downloader
depends on the twitch-python package. Add its dependency.

strager committed:
Twitch-Chat-Downloader: init at 3.2.0

Add Petter Kraabøl's Twitch-Chat-Downloader tool to Nixpkgs, enabling
people to archive Twitch [1] messaging history.

[1] https://www.twitch.tv/

FRidh reviewed (changes requested):
I can think of absolutely *no* good reason why a package would want to import from `pipenv`. 

I had a quick look at the `setup.py` and what is done there is just ridiculous. They use `pipenv ` to create an env, an then use `pipenv` to convert the locked dependencies for `setuptools`.  I get why they do that as it simplifies some of their work but its not what should be done in a `setup.py`. That whole part needs to be patched out before it can go in Nixpkgs.

Pipenv should not be module but remain an application.

strager committed:
pythonPackages.rx: 1.6 -> 3.0.1

Rx version 1.6 is incompatible with Rx version 3.x. I want to add a
package which depends on Rx version 3.x (twitch-python). Upgrade Rx
to version 3.0.1.

From what I can tell, no packages use Rx version 1.6, so this commit
shouldn't break any existing builds (aside from possibly Rx itself).

strager commented on pkgs/tools/networking/twitch-chat-downloader/default.nix:
```diff
@@ -0,0 +1,36 @@
+
+  # Twitch-Chat-Download has no tests at all. Sanity check the command manually.
+  checkPhase = ''
+    "$out"/bin/tcd --help >/dev/null || exit 1
```
Thanks for pointing this out. For some reason, I assumed `-e` was not in effect.

strager commented on pkgs/development/python-modules/rx/3.0.nix:
```diff
@@ -0,0 +1,31 @@
+
+buildPythonPackage rec {
+  pname = "Rx";
+  version = "3.0.1";
```
It looks like nothing uses Rx version 1.6 currently. I'll upgrade it in-place instead of adding a new version.

strager reviewed (commented):
> I can think of absolutely _no_ good reason why a package would want to import from `pipenv`.
> 
> I had a quick look at the `setup.py` and what is done there is just ridiculous. They use `pipenv ` to create an env, an then use `pipenv` to convert the locked dependencies for `setuptools`. I get why they do that as it simplifies some of their work but its not what should be done in a `setup.py`. That whole part needs to be patched out before it can go in Nixpkgs.
> 
> Pipenv should not be module but remain an application.

Good points. Patching the pipenv-using packages was easy, so I'll do that instead of exposing pipenv as a library.

strager commented:
GitHub says "Changes requested", but I made the changes. How do I tell GitHub that I made the requested changes and that I want this PR reviewed?

jonringer commented:
not sure if it shows on your UI, but on main there's a arrow to the right which I can expand and dismiss.

jonringer commented on maintainers/maintainer-list.nix:
```diff
@@ -6432,6 +6432,12 @@
+    github = "strager";
+    githubId = 48666;
+    name = "strager";
+  };
```
please add as separate commit
```
maintainers: add strager
```

jonringer commented on pkgs/tools/networking/twitch-chat-downloader/default.nix:
```diff
@@ -0,0 +1,37 @@
+  meta = with lib; {
+    description = "Download chat messages from past broadcasts on Twitch";
+    license = licenses.mit;
+    homepage = https://github.com/PetterKraabol/Twitch-Chat-Downloader;
```
```suggestion
    homepage = "https://github.com/PetterKraabol/Twitch-Chat-Downloader";
```

jonringer commented on pkgs/top-level/python-packages.nix:
```diff
@@ -1213,7 +1213,7 @@ in {
   rq = callPackage ../development/python-modules/rq { };
 
-  rx = callPackage ../development/python-modules/rx { };
+  rx = disabledIf (! pythonAtLeast "3.6") (callPackage ../development/python-modules/rx { });
```
please move into expression

strager replied:
Will do. I followed the convention set by other packages, such as docker-py, opencv, and meson, but I didn't see that many more packages use `disabled = ...;` instead.

jonringer replied:
the disabledIf works anywhere, but the disabled attr is special to python packages. It's common to have interpreter specific disables, so it's more likely to be commonly used in python

jonringer commented on pkgs/top-level/python-packages.nix:
```diff
@@ -5232,6 +5232,8 @@ in {
 
   twisted = callPackage ../development/python-modules/twisted { };
 
+  twitch-python = disabledIf (! pythonAtLeast "3.7") (callPackage ../development/python-modules/twitch-python { });
```
move into expression please
```suggestion
  twitch-python = disabledIf (! pythonAtLeast "3.7") (callPackage ../development/python-modules/twitch-python { });
```

jonringer commented on pkgs/development/python-modules/twitch-python/default.nix:
```diff
@@ -0,0 +1,29 @@
+
+buildPythonPackage rec {
+  pname = "twitch-python";
+  version = "0.0.17";
```
```suggestion
  version = "0.0.17";
  disabled = pythonOlder "3.6";
```

jonringer replied:
and the other file as well please

jonringer commented:
Made an issue upstream https://github.com/PetterKraabol/Twitch-Python/issues/13

strager committed:
maintainers: add strager

strager commented:
> not sure if it shows on your UI, but on main there's a arrow to the right which I can expand and dismiss.

I think I see the "v" arrow you're talking about. When I click it, a panel opens with the text "FRidh requested changes" and a "..." button. Clicking the "..." button shows a menu with one option: "See review". From there, I don't see any button labelled "dismiss".

jonringer commented:
> > not sure if it shows on your UI, but on main there's a arrow to the right which I can expand and dismiss.
> 
> I think I see the "v" arrow you're talking about. When I click it, a panel opens with the text "FRidh requested changes" and a "..." button. Clicking the "..." button shows a menu with one option: "See review". From there, I don't see any button labelled "dismiss".

![Selection_009](https://user-images.githubusercontent.com/7673602/74902865-346ee180-535c-11ea-80de-859d641cf9b7.png)

this is what mine looks like

jonringer commented on pkgs/tools/networking/twitch-chat-downloader/default.nix:
```diff
@@ -0,0 +1,37 @@
+}:
+
+python3Packages.buildPythonApplication rec {
+  name = "Twitch-Chat-Downloader-${version}";
```
```suggestion
  pname = "Twitch-Chat-Downloader";
```

jonringer reviewed (commented):
otherwise LGTM, assuming everything has some tests

strager commented:
> otherwise LGTM, assuming everything has some tests

I don't think any of these packages have significant tests of their own.

strager commented:
> > > not sure if it shows on your UI, but on main there's a arrow to the right which I can expand and dismiss.
> > 
> > 
> > I think I see the "v" arrow you're talking about. When I click it, a panel opens with the text "FRidh requested changes" and a "..." button. Clicking the "..." button shows a menu with one option: "See review". From there, I don't see any button labelled "dismiss".
> 
> ![Selection_009](https://user-images.githubusercontent.com/7673602/74902865-346ee180-535c-11ea-80de-859d641cf9b7.png)
> 
> this is what mine looks like

I see only "See review":

<img width="922" alt="Screen Shot 2020-02-19 at 23 33 07" src="https://user-images.githubusercontent.com/48666/74910836-43f82580-5370-11ea-8314-9eaec893e7f0.png">

jonringer commented:
https://github.com/PetterKraabol/Twitch-Python/issues/13 was closed, if you want to wait until the next release, and update this, then I think we can have another go

PetterKraabol commented:
Hi, I've removed the pipenv dependecy from setup.py in twitch-python and twitch-chat-downloader.

strager commented:
Gross patch removed. (Thanks @jonringer and @PetterKraabol!) This PR is ready for review again.

strager commented:
graphql-core's build seems to be broken. I'll investigate.

strager committed:
graphql-core: 2.3.1 -> 3.1.0

I want to upgrade rx (a Python library) from version 1.6 to version 3.0.
graphql-core version 2.3.1 requires rx version <2. This prevents me from
upgrading rx.

Upgrade graphql-core to the latest release, removing its dependency on
rx entirely.

Unfortunately, graphql-server-core (which depends on graphql-core) does
not have a release which works with this latest version of graphql-core.
Because graphql-server-core isn't used anywhere yet [1], disable the
package to unblock me upgrading rx.

[1] https://github.com/NixOS/nixpkgs/pull/80828#issuecomment-630508526

FRidh commented:
@GrahamcOfBorg build twitch-chat-downloader

marsam commented:
superseded by https://github.com/NixOS/nixpkgs/pull/112712
